src/pricecypher/contracts/handlers/read_parquet_handler.py

```diff
@@ -0,0 +1,44 @@
+from abc import abstractmethod
+from typing import Any
+
+import pandas as pd
+import pyarrow as pa
+
+from .base_handler import BaseHandler
+from pricecypher.enums import AccessTokenGrantType
+
+
+class ReadParquetHandler(BaseHandler):
+    """The abstract ReadParquetHandler class provides a base to read a parquet file into a pandas DataFrame.
+    Extend this class and override the `process()` method when you want to receive a pyarrow Table to convert to a
+    DataFrame.
+    The input parquet file should be available at the `path_in` location. The output DataFrame will be
+    stored as a pickle at the `path_out` location.
+    """
+
+    def get_allowed_access_token_grant_types(self) -> set[AccessTokenGrantType]:
+        return set()
+
+    def get_config_dependencies(self) -> dict[str, list[str]]:
+        return dict()
+
+    def handle(self, user_input: dict[str, Any]) -> any:
+        """Handle the given `user_input`.
+        Needs a parquet file stored at the `path_in` location. The output pandas DataFrame will be stored as a pickle at
+        the `path_out` location.
+
+        :param user_input: requires `path_in` and `path_out`.
+        :return: the remote storage path.
+        """
+        input_table = self._file_storage.read_parquet(user_input.get('path_in'))
+        output_df = self.process(input_table)
+        return self._file_storage.write_df(user_input.get('path_out'), output_df)
+
+    @abstractmethod
+    def process(self, table: pa.Table) -> pd.DataFrame:
+        """Override with a transformation from a pyarrow Table (read from input parquet file) into a pandas DataFrame.
+
+        :param table: the input pyarrow Table.
+        :return: the resulting DataFrame.
+        """
+        raise NotImplementedError
```

Comment:
Would it make sense to add a default implementation that simply converts/casts the given `table` input into a pandas DataFrame? (i.e. by default acting like the _identity_ function).

Same for the other read and write handlers.

Reply:
I would say yes. Any data transformations should be done in a different step anyway. The only 'weird' thing will be that the main.py file in the workflow step that uses this handler will have a class that extends this handler, but will not override anything. But that is more an issue of the workflow step implementation than an issue of these io handlers.

Reply:
I left the read and write string handlers unchanged, as with string input the format is not certain (csv, json, sql etc). And creating a handler for each pandas read_* method seems a bit much.